hyper/contrib.py

```diff
@@ -43,28 +48,40 @@ def get_connection(self, host, port, scheme, cert=None):
         if cert is not None:
             ssl_context = init_context(cert=cert)
 
+        connection_key = (host, port, scheme, cert, proxy)
```

Comment:
They needn't be. Requests has a `proxy_headers` method expressly to allow it to be overridden, in case users want to provide other forms of proxy header handling.